From 0f330d7488910694296772392b149f2982b9f2e0 Mon Sep 17 00:00:00 2001
From: florent pirou <florent.pirou@intel.com>
Date: Mon, 31 Aug 2026 06:17:38 -0700
Subject: [PATCH 03/80] makefile: disable IPU7 ISYS reset feature

Signed-off-by: florent pirou <florent.pirou@intel.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b2e3ba7b..91fbb2c6 100644
--- a/Makefile
+++ b/Makefile
@@ -87,7 +87,7 @@ LINUXINCLUDE := -I$(src)/include $(LINUXINCLUDE)
 ccflags-y := -I$(src)/include
 # IPU7 driver configs
 export CONFIG_VIDEO_INTEL_IPU7=m
-export CONFIG_VIDEO_INTEL_IPU7_ISYS_RESET=y
+export CONFIG_VIDEO_INTEL_IPU7_ISYS_RESET=n
 export CONFIG_VIDEO_INTEL_IPU6=m
 export CONFIG_VIDEO_INTEL_IPU6_ISYS_RESET=y
 
-- 
2.25.1

