From 8d2fabb52219430a6cbfdaa1a7a64d9a3cfe3a12 Mon Sep 17 00:00:00 2001
From: florent pirou <florent.pirou@intel.com>
Date: Tue, 21 Apr 2026 22:06:22 -0700
Subject: [PATCH 04/80] Makefile: compile ipu7 fw logger support

Signed-off-by: florent pirou <florent.pirou@intel.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 91fbb2c6..9909137b 100644
--- a/Makefile
+++ b/Makefile
@@ -91,7 +91,7 @@ export CONFIG_VIDEO_INTEL_IPU7_ISYS_RESET=n
 export CONFIG_VIDEO_INTEL_IPU6=m
 export CONFIG_VIDEO_INTEL_IPU6_ISYS_RESET=y
 
-subdir-ccflags-$(CONFIG_VIDEO_INTEL_IPU7) += -DCONFIG_VIDEO_INTEL_IPU7 -DIPU8_INSYS_NEW_ABI
+subdir-ccflags-$(CONFIG_VIDEO_INTEL_IPU7) += -DCONFIG_VIDEO_INTEL_IPU7 -DIPU8_INSYS_NEW_ABI -DENABLE_FW_OFFLINE_LOGGER
 subdir-ccflags-$(CONFIG_VIDEO_INTEL_IPU7_ISYS_RESET) += -DCONFIG_VIDEO_INTEL_IPU7_ISYS_RESET
 subdir-ccflags-$(CONFIG_VIDEO_INTEL_IPU6) += -DCONFIG_VIDEO_INTEL_IPU6
 subdir-ccflags-$(CONFIG_VIDEO_INTEL_IPU6_ISYS_RESET) += -DCONFIG_VIDEO_INTEL_IPU6_ISYS_RESET
-- 
2.25.1

