From 84ae41288fae10e6ef78eac6c9f5d609e6b76093 Mon Sep 17 00:00:00 2001
From: Florent Pirou <florent.pirou@intel.com>
Date: Mon, 2 Feb 2026 07:17:45 -0700
Subject: [PATCH 41/80] d4xx-compat: fix dfu char device sysfs naming

Signed-off-by: Florent Pirou <florent.pirou@intel.com>
---
 drivers/media/i2c/d4xx-compat/d4xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/i2c/d4xx-compat/d4xx.c b/drivers/media/i2c/d4xx-compat/d4xx.c
index 558f9fd3..447db269 100644
--- a/drivers/media/i2c/d4xx-compat/d4xx.c
+++ b/drivers/media/i2c/d4xx-compat/d4xx.c
@@ -6456,7 +6456,7 @@ static int ds5_chrdev_init(struct i2c_client *c, struct ds5 *state)
 	 * - IPU6 legacy when standalone <a|b|c|d|e|f> when aggregated <g|h|i|j|k|l|m>
 	 */
 	if (strnlen(spdata->suffix, sizeof(spdata->suffix)) > 1) {
-		snprintf(dev_name, sizeof(dev_name), "D4XX %s %s",
+		snprintf(dev_name, sizeof(dev_name), "%s-%s",
 			 DS5_DRIVER_NAME_DFU, spdata->suffix);
 	}else if (state->aggregated & 1) {
 		suffix += 6;
-- 
2.25.1

