From f65773497504ccd82eb19ffd377caa0468f5504f Mon Sep 17 00:00:00 2001
From: Dmitry Perchanov <dmitry.perchanov@intel.com>
Date: Tue, 13 Aug 2024 16:20:48 +0300
Subject: [PATCH 48/80] d4xx-compat: Locking on open dfu device

---
 drivers/media/i2c/d4xx-compat/d4xx.c | 23 +++++++++++++++++------
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/d4xx-compat/d4xx.c b/drivers/media/i2c/d4xx-compat/d4xx.c
index 42282f92..35a69ccb 100644
--- a/drivers/media/i2c/d4xx-compat/d4xx.c
+++ b/drivers/media/i2c/d4xx-compat/d4xx.c
@@ -6463,16 +6463,20 @@ static int ds5_dfu_device_open(struct inode *inode, struct file *file)
 			state->client->adapter);
 #endif
 #endif
-	if (state->dfu_dev.device_open_count)
+	mutex_lock(&state->lock);
+	if (state->dfu_dev.device_open_count) {
+		mutex_unlock(&state->lock);
 		return -EBUSY;
+	}
 	state->dfu_dev.device_open_count++;
 	if (state->dfu_dev.dfu_state_flag != DS5_DFU_RECOVERY)
 		state->dfu_dev.dfu_state_flag = DS5_DFU_OPEN;
 	state->dfu_dev.dfu_msg = devm_kzalloc(&state->client->dev,
 			DFU_BLOCK_SIZE, GFP_KERNEL);
-	if (!state->dfu_dev.dfu_msg)
+	if (!state->dfu_dev.dfu_msg) {
+		mutex_unlock(&state->lock);
 		return -ENOMEM;
-
+	}
 	file->private_data = state;
 #ifdef CONFIG_TEGRA_CAMERA_PLATFORM
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 10)
@@ -6480,9 +6484,10 @@ static int ds5_dfu_device_open(struct inode *inode, struct file *file)
 	while (parent && i2c_parent_is_i2c_adapter(parent))
 		parent = i2c_parent_is_i2c_adapter(state->client->adapter);
 
-	if (!parent)
+	if (!parent) {
+		mutex_unlock(&state->lock);
 		return 0;
-
+	}
 	dev_dbg(&state->client->dev, "%s(): i2c-%d bus_clk = %d, set %d\n",
 			__func__,
 			i2c_adapter_id(parent),
@@ -6493,6 +6498,7 @@ static int ds5_dfu_device_open(struct inode *inode, struct file *file)
 	i2c_set_adapter_bus_clk_rate(parent, DFU_I2C_BUS_CLK_RATE);
 #endif
 #endif
+	mutex_unlock(&state->lock);
 	return 0;
 };
 
@@ -6556,6 +6562,7 @@ static int ds5_dfu_device_release(struct inode *inode, struct file *file)
 #endif
 #endif
 	int ret = 0, retry = 10;
+	mutex_lock(&state->lock);
 	state->dfu_dev.device_open_count--;
 	if (state->dfu_dev.dfu_state_flag != DS5_DFU_RECOVERY)
 		state->dfu_dev.dfu_state_flag = DS5_DFU_IDLE;
@@ -6573,8 +6580,10 @@ static int ds5_dfu_device_release(struct inode *inode, struct file *file)
 	/* get i2c controller and restore bus clock rate */
 	while (parent && i2c_parent_is_i2c_adapter(parent))
 		parent = i2c_parent_is_i2c_adapter(state->client->adapter);
-	if (!parent)
+	if (!parent) {
+		mutex_unlock(&state->lock);
 		return 0;
+	}
 	dev_dbg(&state->client->dev, "%s(): i2c-%d bus_clk %d, restore to %d\n",
 			__func__, i2c_adapter_id(parent),
 			i2c_get_adapter_bus_clk_rate(parent),
@@ -6592,9 +6601,11 @@ static int ds5_dfu_device_release(struct inode *inode, struct file *file)
 	if (ret) {
 		dev_warn(&state->client->dev,
 			"%s(): no communication with d4xx\n", __func__);
+		mutex_unlock(&state->lock);
 		return ret;
 	}
 	ret = ds5_read(state, DS5_FW_BUILD, &state->fw_build);
+	mutex_unlock(&state->lock);
 	return ret;
 };
 
-- 
2.25.1

