From 5cb501f4e701bfcfb999cb8cd9b5c3178dc81860 Mon Sep 17 00:00:00 2001
From: Arun-Prasad-V <arunprasadv777@gmail.com>
Date: Wed, 30 Oct 2024 20:14:48 +0530
Subject: [PATCH 49/80] d4xx-compat: Locking on mux open and close

---
 drivers/media/i2c/d4xx-compat/d4xx.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/i2c/d4xx-compat/d4xx.c b/drivers/media/i2c/d4xx-compat/d4xx.c
index 35a69ccb..b081fceb 100644
--- a/drivers/media/i2c/d4xx-compat/d4xx.c
+++ b/drivers/media/i2c/d4xx-compat/d4xx.c
@@ -3290,9 +3290,16 @@ static int ds5_mux_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
 	struct ds5 *state = v4l2_get_subdevdata(sd);
 
 	dev_dbg(sd->dev, "%s(): %s (%p)\n", __func__, sd->name, fh);
+
+	mutex_lock(&state->lock);
 	if (state->dfu_dev.dfu_state_flag)
+	{
+		mutex_unlock(&state->lock);
 		return -EBUSY;
+	}
+
 	state->dfu_dev.device_open_count++;
+	mutex_unlock(&state->lock);
 
 	return 0;
 };
@@ -3302,7 +3309,9 @@ static int ds5_mux_close(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
 	struct ds5 *state = v4l2_get_subdevdata(sd);
 
 	dev_dbg(sd->dev, "%s(): %s (%p)\n", __func__, sd->name, fh);
+	mutex_lock(&state->lock);
 	state->dfu_dev.device_open_count--;
+	mutex_unlock(&state->lock);
 	return 0;
 };
 
-- 
2.25.1

